src/kv/kv.h

```diff
@@ -1694,6 +1694,32 @@ namespace kv
           h->append(rep.p, rep.n, data.data(), data.size());
         }
       }
+      else
+      {
+        // Transactions containing a pre prepare or a pbft request should not
+        // contain anything else
```

Comment:
This `else` does not directly correspond to deserialisation of Pre-prepares or Requests but to the fact that we haven't passed a `tx` to `deserialise_views` (i.e. `commit` is `false`). This works for now since this is only used for PBFT but would break if we start using this API for other transactions that contain more than 1 view.

Reply:
I don't understand what you are proposing here. The else will be triggered if we _have_ passed a transaction into deserialise_views and so commit is false. I don't see a use case for pbft where we have more than one view per deserialisation, or a use case in raft where we pass in a transaction or decide to not commit.

Reply:
As discussed we should leave this as is with the consensus assumptions guarding from potential logic bugs and revisit at a later point or when needed (https://github.com/microsoft/CCF/issues/66)